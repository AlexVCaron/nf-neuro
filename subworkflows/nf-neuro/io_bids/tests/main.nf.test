nextflow_workflow {

    name "Test Subworkflow IO_BIDS"
    script "../main.nf"
    workflow "IO_BIDS"

    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/io_bids"

    tag "io"
    tag "io/readbids"

    tag "load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "bids.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("Read BIDS directory - without .bidsignore") {

        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        file("\${test_data_directory}/i_bids", checkIfExists: true)
                    ]
                }
                input[1] = []
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("Read BIDS directory - with .bidsignore") {

        when {
            workflow {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        file("\${test_data_directory}/i_bids", checkIfExists: true)
                    ]
                }
                input[1] = []
                input[2] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        file("\${test_data_directory}/i_bids/.bidsignore")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
