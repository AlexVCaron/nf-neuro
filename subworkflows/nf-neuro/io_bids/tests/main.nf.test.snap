{
    "Read BIDS directory - with .bidsignore": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T02:35:08.920096313"
    },
    "Read BIDS directory - without .bidsignore": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-06T02:35:01.267035439"
    }
}